File: js/src/samples/verification/verification.ts
import { Bloock, BloockClient, Network, RecordBuilder } from "@bloock/sdk";
import { Config } from "../../utils/config";
import { Sample } from "../../utils/sample";
import { Logger } from "../../utils/logger";

Sample.run("Verification", async (config: Config) => {
  // we set the API key and create a client
  Bloock.setApiKey(config.apiKey);
  let client = new BloockClient();

  let record = await RecordBuilder.fromString("Hello world").build();

  let records = [record];

  let receipts = await client.sendRecords(records);

  // Once we sent a record, we can wait for it's anochor
  Logger.info(`Waiting for anchor...`);
  // we can optionally specify a timeout (if not set, default is 120000)
  let anchor = await client.waitAnchor(receipts[0].anchor, 120000);
  Logger.success(`Anchor ${JSON.stringify(anchor)} done!`);

  // we can optionally specify a network (if not set, default is Ethereum Mainnet)
  let timestamp = await client.verifyRecords(records, Network.BLOOCK_CHAIN);
  Logger.success(`Timestamp: ${timestamp}`);
});

File: java/src/main/java/com/bloock/quickstart/samples/verification/Verification.java
package com.bloock.quickstart.samples.verification;

import com.bloock.quickstart.utils.Config;
import com.bloock.quickstart.utils.Logger;
import com.bloock.quickstart.utils.Sample;
import com.bloock.sdk.Bloock;
import com.bloock.sdk.builder.Builder;
import com.bloock.sdk.client.Client;
import com.bloock.sdk.entity.Anchor;
import com.bloock.sdk.entity.Network;
import com.bloock.sdk.entity.Record;
import com.bloock.sdk.entity.RecordReceipt;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Verification extends Sample {
  public void run(Config config) throws Exception {
    Bloock.apiKey = config.apiKey;
    Client client = new Client();

    Record record = Builder.fromString("Hello world").build();

    String hash = record.getHash();
    ArrayList<String> records = new ArrayList<>(Arrays.asList(hash));

    List<RecordReceipt> receipts = client.sendRecords(records);

    // Once we sent a record, we can wait for it's anochor
    Logger.info("Waiting for anchor...");
    // we can optionally specify a timeout (if not set, default is 120000)
    Anchor anchor = client.waitAnchor(receipts.get(0).getAnchor(), 120000);
    Logger.success("Anchor " + anchor.getId() + " done!");

    // we can optionally specify a network (if not set, default is Ethereum Mainnet)
    long timestamp = client.verifyRecords(records, Network.BLOOCK_CHAIN);
    Logger.success("Timestamp: " + timestamp);
  }

  Verification(String name) {
    super(name);
  }

  public static void main(String[] args) {
    new Verification("Verification");
  }
}
